eth/handler.go

```diff
@@ -170,6 +171,12 @@ func newHandler(config *handlerConfig) (*handler, error) {
 	heighter := func() uint64 {
 		return h.core.CurrentHeader().NumberU64()
 	}
+	currentThresholdS := func() *big.Int {
+		return h.core.Engine().IntrinsicLogS(h.core.CurrentHeader().Hash())
+	}
+	currentS := func() *big.Int {
+		return h.core.CurrentHeader().ParentEntropy()
```

Comment:
Can you explain why parent entropy of CurrentHeader instead of just entropy of CurrentHeader?

Reply:
We just need to approximate here as computing the CurrentEntropy using the ParentEntropy and ParentDeltaS arrays and also computing intrinsic and adding them all together on every single broadcast is very expensive

Reply:
Lets add a comment for the above rational

Reply:
Sounds good, yes our range will be off by one block. Worst case is everyone is right on the boundary and half accept and half reject, but eventually the one with more hash power will win.

Reply:
Could you please add a comment to indicate this is an approximate off by one? Other people may copy this code and it will be incorrect if used anywhere else.

Reply:
we are talking about a boundary 3500 blocks away

Reply:
> Could you please add a comment to indicate this is an approximate off by one? Other people may copy this code and it will be incorrect if used anywhere else.

Or could call it approximateCurrentS or approximateS or something.

Reply:
I will add comments, I think introducing new terms is probably confusing at this point